app/jobs/onceoff/rebake_gif_images.rb

```diff
@@ -0,0 +1,43 @@
+# frozen_string_literal: true
+
+module Jobs
+  class RebakeGifImages < ::Jobs::Onceoff
+    def execute_onceoff(args)
+      Upload.where("original_filename LIKE '%.gif'").find_each do |upload|
+        uri = Discourse.store.path_for(upload) || upload.url
+        upload.update!(animated: FastImage.animated?(uri))
+      end
+
+      User
+        .joins(:uploaded_avatar)
+        .where("uploads.original_filename LIKE '%.gif'")
+        .find_each do |user|
+
+        if user.uploaded_avatar.animated
+          if path = Discourse.store.path_for(user.uploaded_avatar)
+            file = File.new(path)
+          else
+            find = Discourse.store.download(user.uploaded_avatar)
```

Comment:
I think you meant `file` here?

```suggestion
            file = Discourse.store.download(user.uploaded_avatar)
```

Reply:
Yes, you are right. 🤦🏻‍♀️